service-pdf/tests/OpgTest/Lpa/Pdf/Config/ConfigTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace OpgTest\Lpa\Pdf\Config;
+
+use Opg\Lpa\Pdf\Config\Config;
+use PHPUnit\Framework\TestCase;
+
+class ConfigTest extends TestCase
+{
+    public function testGetInstance()
+    {
+        $configObj1 = Config::getInstance();
+        $configObj2 = Config::getInstance();
+
+        $this->assertEquals(spl_object_hash($configObj1), spl_object_hash($configObj2));
+    }
+
+    public function testOffsetSetOffsetGet()
+    {
+        $some = [
+            'new' => [
+                'config' => true,
+            ]
+        ];
+
+        $configObj = Config::getInstance();
+
+        $configObj->offsetSet('some-new-config', $some);
+
+        $this->assertEquals($configObj->offsetGet('some-new-config'), $some);
+    }
+
+    public function testOffsetSetNoKey()
+    {
+        $some = [
+            'other' => [
+                'config' => false,
+            ]
+        ];
+
+        $configObj = Config::getInstance();
+
+        $configObj->offsetSet(null, $some);
+
+        $this->assertEquals($configObj->offsetGet(0), $some);
+    }
+
+    public function testOffsetExists()
+    {
+        $configObj = Config::getInstance();
+
+        $this->assertFalse($configObj->offsetExists('ghi'));
+
+        $configObj->offsetSet('ghi', 'some-text-string');
+
+        $this->assertTrue($configObj->offsetExists('ghi'));
+    }
+
+    public function testOffsetUnset()
+    {
+        $configObj = Config::getInstance();
+
+        $this->assertFalse($configObj->offsetExists('jkl'));
+
+        $configObj->offsetSet('jkl', 12345);
+
+        $this->assertTrue($configObj->offsetExists('jkl'));
+
+        $configObj->offsetUnset('jkl');
+
+        $this->assertFalse($configObj->offsetExists('jkl'));
+    }
+
+    public function testCount()
+    {
+        $configObj = Config::getInstance();
+        $configObjCount = $configObj->count();
+
+        $configObj->offsetSet('abc', true);
+        $configObj->offsetSet('def', false);
+
+        $this->assertEquals($configObj->count(), $configObjCount + 2);
+    }
+
+    public function testMerge()
+    {
+        $data1 = [
+            'data1' => true,
+            'data2' => null,
+            'data3' => false,
+            5       => 'something',
+        ];
+
+        $data2 = [
+            'data1' => true,
+            'data2' => 'abc',
+            'data4' => 987,
+            5       => 'else',
```

Comment:
Should this value be `6` instead of `5`? Going by the assert listed below.

Reply:
According to the code, it shouldn't: the default behaviour for numeric keys (unless you pass `$preserveNumericKeys=true` to the `merge()` function, is to append the value to the list (which will assign it a new key). This seems unintuitive and not pleasant, but I'm not going to change it now as it may have knock-on effects elsewhere.

Reply:
I thought that might be the case, but you second guess yourself from looking at the code.